app/controllers/document_topics_controller.rb

```diff
@@ -12,5 +12,18 @@ class DocumentTopicsController < ApplicationController
   def edit
     @document = Document.find_by_param(params[:document_id])
     @topic_index = TopicsService.new.topic_index
+    @topic_content_ids, @version = TopicsService.new.topics_for_document(@document)
+  end
+
+  def update
+    document = Document.find_by_param(params[:document_id])
+    TopicsService.new.patch_topics(document, params.fetch(:topics, []), params[:version])
+    redirect_to document_path(document), notice: t("documents.show.flashes.topics_updated")
+  rescue GdsApi::HTTPConflict => e
+    Rails.logger.error(e)
+    redirect_to document, alert_with_description: t("documents.show.flashes.topic_update_conflict")
```

Comment:
Should we be redirecting them to the topic edit page here? seems a bit of a bummer to ask them to go back to the form when we could take them there.

Reply:
Can do - my thought was to be consistent with the 'API down' case. But actually Whitehall does this, so we may as well behave the same - @MuriloDalRi would you like to change this?

Reply:
Done this :-)